### PR TITLE
Create FM_BLOCKCYCLER.sma

### PR DESCRIPTION
**Pull request by MrCoala**
Decompiled the plugin of orion and tried to reconstruct the code. Description: Blocks USE and ATTACK on cyclers, making them not cycle animations.

@VANCOLD and @the-remeko I am not sure if this plugin will work this way since at line 39 for example RegisterHam is called on value 43 which i assume is the offset for Use? The offset could change any time i guess? Cant we use the external config?

Please check the code and test it, i wasnt able to.

### CODE REVIEW

**Comment by the-remeko** [comment]
@MrCoala I removed the spawn hook used to disable damage on the entity and just registered a TakeDamage hook that cancels just like the Use hook. The code is much simpler now, but please advise if hooking into spawn would be more stable or if doing it differently could cause side effects.

**Comment by VANCOLD** [comment]
hooking into spawn doesn't cause any side effects, you are simply blocking a interaction. Hooking into spawn of an entity is usually only needed if you need to change values and not alter behaviour.
that's why set_pev was used in the old spawn to change a value. Since it would we easier to just block out the behaviour imo with he current setup this also fixes he issue.
=> hooking into spawn means setting values usually or block spawning 
=> hooking into behaviour means you only alter the behaviour

**Review by MrCoala — COMMENTED** [review]
I really like the way you solved it. @VANCOLD can you review?

File: FM_BLOCKCYCLER.sma [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include <amxmodx>
+#include <hamsandwich>
+
+public plugin_init()
+{
+    register_plugin("FM_BLOCKCYCLER", "1.1", "teh ORiON");
```
Comment (VANCOLD) [inline FM_BLOCKCYCLER.sma:6]: bro you can give yourself some credit

File: FM_BLOCKCYCLER.sma [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    RegisterHam(Ham_Use, "cycler", "Block");
+    RegisterHam(Ham_TakeDamage, "cycler", "Block");
+
+    return 0;
```
Comment (VANCOLD) [inline FM_BLOCKCYCLER.sma:11]: please replace return 0 with PLUGIN_CONTINUE 
see https://www.amxmodx.org/api/amxconst#return-codes

File: FM_BLOCKCYCLER.sma (hunk 1 above)
Comment (VANCOLD) [reply FM_BLOCKCYCLER.sma]: - sidenote: fm_RegisterPlugin instead of register_plugin()

**Review by VANCOLD — APPROVED** [review]
looks fine now

**Pull request merged**